code/ir/rpc.py

```diff
@@ -1 +1,50 @@
-rpc.py
\ No newline at end of file
+import socket
+
+EOM = '---EOM'
+
+
+def start_ir(host, port, ir, max_connections=1):
+    # Setup socket
+    s = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+    s.bind((host, port))
+    s.listen(max_connections)
+    conn, addr = s.accept()
+
+    # Keep on listening
+    while True:
+        # accept new message
+        newsock, address = s.accept()
+
+        # receive questions from new connection
+        message = recv_end(newsock)
+        type, question = message.split(" ", 1)
+        answers = ir.question(type, question)
+
+        # format question
+        formatted_answers = "\n---\n".join(
+            ["\n".join([str(a[0]), a[1], a[2]])
+                for a in answers]) + "\n---EOM"
+        conn.send(formatted_answers)
+
+
+# this makes sure that we receive a question in the following form:
+#    any message here as long as it ends with---EOM
+def recv_end(the_socket):
+    total_data = []
+    data = ''
+    while True:
+        data = the_socket.recv(8192)
```

Comment:
No, I could set that to be anything.
That is a good estimate so that I can more or less get any possible answer in one send only, instead of multiple